CDC-Data-Reconciliation-Backend/server.py

```diff
@@ -123,6 +124,51 @@ async def manual_report(state_file: UploadFile = File(None), cdc_file:  UploadFi
     os.remove(res_file)
     os.rmdir(os.path.join(app.dir, folder_name, id))
 
+    currTime = datetime.now()
+    createdAtDate = currTime.strftime("%Y-%m-%d")
+    timeOfCreation = currTime.strftime("%H:%M:%S")
+
+    numDiscrepancies = len(res)
+    reportId = insert_report(createdAtDate, timeOfCreation, numDiscrepancies)
+    
+    statistics_summary = {}
+      
+    for case in res:
+        eventCode = case["EventCode"]
+        reasonID = case["ReasonID"]
+        
+        # Initialize the event code entry if not present
+        if eventCode not in statistics_summary:
+            statistics_summary[eventCode] = {
+                "TotalCases": 0,
+                "TotalDuplicates": 0,
+                "TotalMissingFromCDC": 0,
+                "TotalMissingFromState": 0,
+                "TotalWrongAttributes": 0,
+            }
+        
+        statistics_summary[eventCode]["TotalCases"] += 1
+
+        if reasonID == 1:  
+            statistics_summary[eventCode]["TotalDuplicates"] += 1
+        elif reasonID == 2: 
+            statistics_summary[eventCode]["TotalMissingFromCDC"] += 1
+        elif reasonID == 3: 
+            statistics_summary[eventCode]["TotalWrongAttributes"] += 1
+        elif reasonID == 4: 
+            statistics_summary[eventCode]["TotalMissingFromState"] += 1
+
+    for event_code, stats in statistics_summary.items():
+        insert_statistics(
+            reportId,
+            event_code,
+            stats["TotalCases"],
+            stats["TotalDuplicates"],
+            stats["TotalMissingFromCDC"],
+            stats["TotalMissingFromState"],
+            stats["TotalWrongAttributes"],
+        )
+
```

Comment:
Do no calculate statistics here. You do not know the total amount of cases analyzed here and thus this is why John is calculating the statistics in compare.py.